src/PowerShellEditorServices.Hosting/Configuration/TransportConfig.cs

```diff
@@ -0,0 +1,185 @@
+﻿//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.IO.Pipes;
+using System.Threading.Tasks;
+
+namespace Microsoft.PowerShell.EditorServices.Hosting
+{
+    /// <summary>
+    /// Configuration specifying an editor services protocol transport stream configuration.
+    /// </summary>
+    public interface ITransportConfig
+    {
+        /// <summary>
+        /// Create, connect and return the configured transport streams.
+        /// </summary>
+        /// <returns>The connected transport streams. inStream and outStream may be the same stream for duplex streams.</returns>
+        Task<(Stream inStream, Stream outStream)> ConnectStreamsAsync();
+
+        /// <summary>
+        /// The name of the transport endpoint for logging.
+        /// </summary>
+        string EndpointDetails { get; }
+
+        /// <summary>
+        /// The name of the transport to record in the session file.
+        /// </summary>
+        string SessionFileTransportName { get; }
+
+        /// <summary>
+        /// Extra entries to record in the session file.
+        /// </summary>
+        IReadOnlyDictionary<string, object> SessionFileEntries { get; }
+    }
+
+    /// <summary>
+    /// Configuration for the standard input/output transport.
+    /// </summary>
+    public class StdioTransportConfig : ITransportConfig
+    {
+        public string EndpointDetails => "<stdio>";
+
+        public string SessionFileTransportName => "Stdio";
+
+        public IReadOnlyDictionary<string, object> SessionFileEntries { get; } = null;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this initialization to 'SessionFileEntries', the compiler will do that for you.](https://app.codacy.com/app/TylerLeonhardt/PowerShellEditorServices/pullRequest?prid=4579423)